references/cuegen/util.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2023 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cuegen
+
+import (
+	"fmt"
+	gotypes "go/types"
+	"strconv"
+	"unicode"
+	"unicode/utf8"
+
+	cueast "cuelang.org/go/cue/ast"
+	cuetoken "cuelang.org/go/cue/token"
+)
+
+var _ = ident("", true) // nolint. will be used in the future. TODO(iyear): remove
+
+func ident(name string, isDef bool) *cueast.Ident {
+	if isDef {
+		// get the first rune
+		r, _ := utf8.DecodeRuneInString(name)
+		name = "#" + name
+		if !unicode.Is(unicode.Lu, r) {
+			name = "_" + name
+		}
+	}
+	return cueast.NewIdent(name)
+}
+
+func basicType(x *gotypes.Basic) cueast.Expr {
+	// byte is an alias for uint8 in go/types
+
+	switch t := x.String(); t {
+	case "uintptr":
+		return ident("uint64", false)
+	case "byte":
+		return ident("uint8", false)
```

Comment:
Why not `bytes`?

Reply:
This is the base type conversion, so it just convert byte in Golang to uint8 in CUE(CUE don't have `byte` type, only `bytes`). 

And `[]byte` and `[...]byte` are processed at https://github.com/kubevela/kubevela/pull/5585/files#diff-f3ec0bb030f3add13296ed258157b227f061b722669287e2ad2cc069d76e2723R101 and https://github.com/kubevela/kubevela/pull/5585/files#diff-f3ec0bb030f3add13296ed258157b227f061b722669287e2ad2cc069d76e2723R110